arangod/Aql/IResearchViewNode.cpp

```diff
@@ -1337,53 +1342,110 @@ IResearchViewNode::IResearchViewNode(aql::ExecutionPlan& plan,
+            static_cast<size_t>(_heapSort.back().source) >= _scorers.size()) {
           THROW_ARANGO_EXCEPTION_MESSAGE(
               TRI_ERROR_BAD_PARAMETER,
-              absl::StrCat("'scorersSort[", itr.index(),
-                           "].index' attribute is out of range"));
+              absl::StrCat("'heapSort[", itr.index(),
+                           "].source' attribute is out of range"));
         }
-        _scorersSort.emplace_back(index.getNumber<size_t>(), asc.getBool());
       } else {
         THROW_ARANGO_EXCEPTION_MESSAGE(
-            TRI_ERROR_BAD_PARAMETER, absl::StrCat("'scorersSort[", itr.index(),
+            TRI_ERROR_BAD_PARAMETER, absl::StrCat("'heapSort[", itr.index(),
                                                   "]' attribute is invalid "));
       }
     }
-  }
-  auto const scorersSortLimitSlice = base.get(NODE_VIEW_SCORERS_SORT_LIMIT);
-  if (!scorersSortLimitSlice.isNone()) {
-    if (!scorersSortLimitSlice.isNumber()) {
-      THROW_ARANGO_EXCEPTION_MESSAGE(
-          TRI_ERROR_BAD_PARAMETER,
-          "'scorersSortLimit' attribute should be a numeric");
+    auto const heapSortLimitSlice = base.get(kNodeViewHeapSortLimit);
+    if (!heapSortLimitSlice.isNone()) {
+      if (!heapSortLimitSlice.isNumber()) {
+        THROW_ARANGO_EXCEPTION_MESSAGE(
+            TRI_ERROR_BAD_PARAMETER,
+            "'heapSortLimit' attribute should be a numeric");
+      }
+      _heapSortLimit = heapSortLimitSlice.getNumber<size_t>();
+    }
+  } else {
+    // TODO remove when 3.11 is deprecated
+    auto const scorersSortSlice = base.get(kNodeViewScorersSort);
+    if (!scorersSortSlice.isNone()) {
+      if (!scorersSortSlice.isArray()) {
+        THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_BAD_PARAMETER,
+                                       "'scorersSort' should be an array");
+      }
+      auto itr = velocypack::ArrayIterator(scorersSortSlice);
+      for (auto const scorersSortElement : itr) {
+        if (!scorersSortElement.isObject()) {
+          THROW_ARANGO_EXCEPTION_MESSAGE(
+              TRI_ERROR_BAD_PARAMETER,
+              absl::StrCat("'scorersSort[", itr.index(),
+                           "]' attribute should be an object"));
+        }
+        auto index = scorersSortElement.get(kNodeViewScorersSortIndex);
+        auto asc = scorersSortElement.get(kNodeViewScorersSortAsc);
+        if (index.isNumber() && asc.isBoolean()) {
+          auto indexVal = index.getNumber<size_t>();
+          if (indexVal >= _scorers.size()) {
+            THROW_ARANGO_EXCEPTION_MESSAGE(
+                TRI_ERROR_BAD_PARAMETER,
+                absl::StrCat("'scorersSort[", itr.index(),
+                             "].index' attribute is out of range"));
+          }
+          _heapSort.emplace_back(
+              HeapSortElement{.postfix = "",
+                              .source = index.getNumber<ptrdiff_t>(),
+                              .fieldNumber = std::numeric_limits<size_t>::max(),
```

Comment:
could we not init default initted fields?

Reply:
I tries. CI was not happy.